auth/auth.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def implicit project_id:
+  # [START implicit]
+  # project_id = "Your Google Cloud project ID"
+
+  require "google/cloud/storage"
+
+  # If you don't specify credentials when constructing the client, the client
+  # library will look for credentials in the environment.
```

Comment:
Understood, follow-up response:

* This comment is used by other language samples ([Python](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/auth/cloud-client/snippets.py#L24) and [Node](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/auth/auth.js#L28)).

* The environment variable is documented on the page https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application. I didn't want to repeat the same wording in the sample to reflect a similar sample used by other languages.